### ISSUE TITLE
General logging of the sysyem

### ISSUE THREAD

**Issue by DonLexos** [pi-hole/pi-hole#340]
Right now the system logs a few things. After an installation there is an installation log, and during the day all activity from dnsmasq is logged. In the default installation that logfile is purged, therefor the admin interface will only show 'todays' activity.

Would it be helpfull if there is a general logfile that is not overwritten and saves a few things? Think things like the weekly update of the adlists, the automatic update of the system, maybe even the daily statistics before the normal pihole log is flushed at the end of the day (`logfile flushed, results for 2016-02-16: xx queries, xx.xxx dns queries, xx % ads, on this day xx.xxx domains are blocked by pi-hole`).

That way you can look back and see if things have indeed taken place. It would also allow for things like `pi-hole system upgraded from 2.5.2 to 2.5.3 on 2016-12-18 23:55 hrs`.
##

**Comment by jradwan** [comment]
The scheduled cron jobs (/etc/cron.d/pihole) take care of some of this if you've got your system configured to send e-mails for cron tasks. I get an e-mail every Sunday morning when gravity.sh runs, and I added a task to run chronometer.sh before the piholeLogFlush.sh nightly job so I get an e-mail with the day's statistics.

It looks like there's a placeholder in the cron file for a dailyLog.sh script for which the comment says "Parse the log file before it is flushed and save the stats to a database" so they apparently have plans for daily logging at some point.

**Comment by PromoFaux** [comment]
> so they apparently have plans for daily logging at some point.

Soon™

**Comment by DonLexos** [comment]
I know it's there, I even tried the MySQL version of the admin (slick!) and atempted to port it to SQLite but failed. I was talking more in the sense of the ability to check if things worked, not the Admin LTE interface which I think (or hope) will be extended over time. Some of the functionality would require some sort of database since all stats are purged at 23:50 hrs to reset it for the next day.

I don't believe anything (of importance) is logged to the `/var/log/messages` which could also be the location. It is more that I thought it could be nice to have a logfile of events that can be reviewed to check if certain events did or did not happen (and worked/failed/etc). Could also help in bughunting (provided elements of the installation log are also kept there). Right now the logfile is overwritten (I think) each week after the automated update script.

**Comment by DkYSwe** [comment]
About logging, it would be nice to have the data for all DNS lookups for the X latest days where X is a configuration parameter.
If you have a Network Intrusion Detection System (NIDS), some of the alerts trigger on part of the domain name but you don't get all the full FQDN, it's here another useful feature of pi-hole comes in. :)

**Comment by jacobsalmela** [comment]
Closing this and [sending it into orbit](https://discourse.pi-hole.net/t/general-logging-of-the-system/552) so this feature is votable.
